client/post-editor/editor-sidebar/header.jsx

```diff
@@ -23,56 +25,74 @@ import DraftsButton from 'post-editor/drafts-button';
 import PostCountsData from 'components/data/post-counts-data';
 import QueryPostTypes from 'components/data/query-post-types';
 
-function EditorSidebarHeader( { typeSlug, type, siteId, showDrafts, toggleDrafts, allPostsUrl, toggleSidebar } ) {
-	const isCustomPostType = ( 'post' !== typeSlug && 'page' !== typeSlug );
-	const className = classnames( 'editor-sidebar__header', {
-		'is-drafts-visible': showDrafts,
-		'is-loading': isCustomPostType && ! type
-	} );
-	const closeLabel = translate( 'Back' );
-	const closeButtonAction = page.back.bind( page, allPostsUrl );
-	const closeButtonUrl = '';
-	const closeButtonAriaLabel = translate( 'Go back' );
+const EditorSidebarHeader = React.createClass( {
```

Comment:
Could you explain the rationale for moving this from stateless function component to a `React.createClass` object? If it's for the pure-render mixin, note that react-redux `connect` includes this functionality in its wrapping behavior ([docs](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)).